include/bm/bm_sim/match_units.h

```diff
@@ -50,11 +50,14 @@ struct MatchKeyParam {
   // VALID used to be before RANGE, but when RANGE support was added, it was
   // easier (implementation-wise) to put RANGE first. Note that this order only
   // matters for the implementation.
+  // TODO(jafingerhut): Where should OPTIONAL be added in this
```

Comment:
You just need to put OPTIONAL with TERNARY here since they are implemented the same way. I suggest putting OPTIONAL *after* TERNARY and replacing this TODO comment with something like: "OPTIONAL is just a TERNARY match where the mask is constrained to be one of 2 values (all 0s or all 1s), so it is implemented just like TERNARY. The 2 match types should be listed one after the other in the enum definition."